src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -683,6 +640,43 @@ public boolean test(Throwable throwable) {
 			return false;
 		}
 
+		@Override
+		public Context currentContext() {
+			return sink.currentContext();
+		}
+
+		@Override
+		public void onUncaughtException(Connection connection, Throwable error) {
+			if (error instanceof RedirectClientException) {
+				if (log.isDebugEnabled()) {
+					log.debug(format(connection.channel(), "The request will be redirected"));
+				}
+			}
+			else if (AbortedException.isConnectionReset(error) && !retried) {
```

Comment:
basically it looks like a special case of error. I might say this is like doOnError, except we know it will be retried, then pass the `error` to it.

Code-wise, brave/zipkin will do nothing different that the hook is named doOnRetry vs doOnError as the behavior for any follow-up request would be the same, regardless of if reason is retry or redirect.